pkg/manager-server/file_system_xfs.go

```diff
@@ -66,6 +67,32 @@ func (f *FileSystemXfs) Create(devices []string, opts FileSystemOptions) error {
 		return err
 	}
 
+	userID := 0
```

Comment:
Since the new code is the same for GFS2/XFS, both which leverage LVM, belongs as a helper method in file_system_lvm.go.

What about "raw" == lvm?

Reply:
I agree that it would be nice to have it as a helper function. I couldn't think of a place to put it, though. file_system_lvm.go doesn't make sense to me because the mount/chown/unmount operations are all done on the file system on top of the LVM volume. "raw" wouldn't use this code because there's no file system to mount. You're more familiar with the code layout here, so if you think that's the best place for it, I can put it there.

Reply:
I think a function makes sense. It doesn't need to be part of any specific file system and instead can use the FileSystemAPI

```
func setFileSystemPermissions(f FileSystemApi, opts FileSystemOptions) (err error) {

        // I put these here since the exports are not used in nnf-sos
	const (
		UserID = "userID"
		GroupID = "groupID"
	)
	
	userID := 0
	if _, exists := opts[UserID]; exists {
		userID = opts[UserID].(int)
	}

	groupID := 0
	if _, exists := opts[GroupID]; exists {
		groupID = opts[GroupID].(int)
	}

	mountpath := "/mnt/nnf/client/" + f.Name()
	if err := f.Mount(mountpath); err != nil {
		return err
	}

	defer func() {
		unmountErr := f.Unmount(mountpath)
		if err == nil {
			err = unmountErr
		}
	}()

	if err := os.Chown(mountpath, userID, groupID); err != nil {
		return err
	}

	return nil
}
```